compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -341,7 +341,7 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
     /// Report an error because the universal region `fr` was required to outlive
     /// `outlived_fr` but it is not known to do so. For example:
     ///
-    /// ```compile_fail,E0312
+    /// ```ignore(compile_fail,E0312)
```

Comment:
Why was this change necessary?

Reply:
Without `ignore`, CI would have failed on doctest execution.
Line 345 seems to be an example written with the intention of causing a compilation error.

Reply:
I don't understand. The whole point of `compile_fail` is that it expects a compilation error, and fails unless that error was present. Why was it necessary to switch from `compile_fail` to `ignore`?

Reply:
@jyn514 
Maybe I'm missing something, but it's because I get the following error.
```shell
# python3 x.py test compiler/rustc_borrowck --stage 0
Building rustbuild
    Finished dev [unoptimized] target(s) in 0.17s
Building stage0 std artifacts (aarch64-apple-darwin -> aarch64-apple-darwin)
    Finished release [optimized] target(s) in 0.16s
Copying stage0 std from stage0 (aarch64-apple-darwin -> aarch64-apple-darwin / aarch64-apple-darwin)
Testing ["rustc_borrowck"] stage0 (aarch64-apple-darwin -> aarch64-apple-darwin)
   Compiling rustc_borrowck v0.0.0 (/Users/reez12g/development/rust/compiler/rustc_borrowck)
    Finished release [optimized] target(s) in 3.24s
     Running unittests src/lib.rs (build/aarch64-apple-darwin/stage0-rustc/aarch64-apple-darwin/release/deps/rustc_borrowck-b5f47d79c70b6374)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests rustc_borrowck

running 14 tests
iFFiiFi.......
failures:

---- src/diagnostics/region_errors.rs - diagnostics::region_errors::MirBorrowckCtxt::report_region_error (line 336) stdout ----
error: lifetime may not live long enough
 --> src/diagnostics/region_errors.rs:337:41
  |
3 | fn foo<'a, 'b>(x: &'a u32) -> &'b u32 { x }
  |        --  --                           ^ function was supposed to return data with lifetime `'b` but it is returning data with lifetime `'a`
  |        |   |
  |        |   lifetime `'b` defined here
  |        lifetime `'a` defined here
  |
  = help: consider adding the following bound: `'a: 'b`

error: aborting due to previous error

Some expected error codes were not found: ["E0312"]
---- src/region_infer/mod.rs - region_infer::RegionInferenceContext::check_universal_regions (line 1403) stdout ----
error: lifetime may not live long enough
 --> src/region_infer/mod.rs:1404:41
  |
3 | fn foo<'a, 'b>(x: &'a u32) -> &'b u32 { x }
  |        --  --                           ^ function was supposed to return data with lifetime `'b` but it is returning data with lifetime `'a`
  |        |   |
  |        |   lifetime `'b` defined here
  |        lifetime `'a` defined here
  |
  = help: consider adding the following bound: `'a: 'b`

error: aborting due to previous error

Some expected error codes were not found: ["E0312"]
---- src/region_infer/mod.rs - region_infer::RegionInferenceContext::check_polonius_subset_errors (line 1458) stdout ----
error: lifetime may not live long enough
 --> src/region_infer/mod.rs:1459:41
  |
3 | fn foo<'a, 'b>(x: &'a u32) -> &'b u32 { x }
  |        --  --                           ^ function was supposed to return data with lifetime `'b` but it is returning data with lifetime `'a`
  |        |   |
  |        |   lifetime `'b` defined here
  |        lifetime `'a` defined here
  |
  = help: consider adding the following bound: `'a: 'b`

error: aborting due to previous error

Some expected error codes were not found: ["E0312"]

failures:
    src/diagnostics/region_errors.rs - diagnostics::region_errors::MirBorrowckCtxt::report_region_error (line 336)
    src/region_infer/mod.rs - region_infer::RegionInferenceContext::check_polonius_subset_errors (line 1458)
    src/region_infer/mod.rs - region_infer::RegionInferenceContext::check_universal_regions (line 1403)

test result: FAILED. 7 passed; 3 failed; 4 ignored; 0 measured; 0 filtered out; finished in 1.73s

error: test failed, to rerun pass '--doc'
	finished in 6.097 seconds
Build completed unsuccessfully in 0:00:08
```

But I understand what you are saying now. Surely we should write `compile_fail` here instead of `ignore` (I didn't know about `compile_fail`).

I haven't looked into this closely yet, but is this an error specific to E0312?
It does not seem to occur for other codes, e.g. E0515(not sure).
https://github.com/rust-lang/rust/blob/master/compiler/rustc_borrowck/src/type_check/mod.rs#L976

Reply:
Correct - looks like that error doesn't have a code for some reason. Removing `E0312` from the doctests lang string for now seems fine.